src/sscg_generator_cli.erl

```diff
@@ -33,83 +32,84 @@ set_log_level(#{verbose := 1}) ->
 set_log_level(#{}) ->
     ok.
 
+-doc "Aborts the program with a red formatted error message".
+-spec abort(Format :: string(), Args :: list()) -> Result :: no_return().
 abort(Format, Args) ->
     io:format("~s~n", [color:red(io_lib:format(Format, Args))]),
     erlang:halt(1).
 
+-doc #{equiv => print(Text, [])}.
 print(Text) -> print(Text, []).
+
+-doc "Prints a formatted message to the console".
+-spec print(string(), [term()]) -> ok.
 print(Format, Args) ->
     io:format(Format ++ "~n", Args).
 
+-doc "Prompts the user for input".
+-spec input(string()) -> binary().
 input(Prompt) ->
     case io:get_line(Prompt ++ " ") of
         eof ->
             "";
         {error, Reason} ->
             abort("Error reading input: ~p", [Reason]);
         Data ->
-            re:replace(Data, "^[[:space:]]*+|[[:space:]]*+$", <<>>,
-                [global, {return, binary}]
-            )
+            re:replace(Data,
+                       "^[[:space:]]*+|[[:space:]]*+$",
+                       <<>>,
+                       [global, {return, binary}])
     end.
 
+-doc "Prompts a yes/no-question and returns a boolean based on the response".
+-spec confirm(string()) -> boolean().
 confirm(Prompt) ->
     case re:run(input(Prompt), ?YES, [{capture, none}]) of
         match -> true;
         _     -> false
     end.
 
-parse_range(Range, {DefaultFrom, DefaultTo}) ->
-    {From, To} = case string:split(Range, "..") of
-        [Start, []] ->
-            {parse_date(Start, first), DefaultTo};
-        [[], ToString] ->
-            {DefaultFrom, parse_date(ToString, last)};
-        [Start, ToString] ->
-            {parse_date(Start, first), parse_date(ToString, last)};
-        _ ->
-            error(invalid_argument)
-    end,
-    case {From, To} of
-        {From, To} when From > To -> error(invalid_argument);
-        % {_, To} when To > DefaultTo -> error(invalid_argument);
-        Else -> Else
+-doc """
+Parses a binary containing a comma-separated list of authors into a list of maps.
+Each author entry should be in the format "<name>:<email>". Entries with missing
+ name or email are also supported:
+- "<name>:" will result in a map with `name` set and `email` as `undefined`.
+- ":<email>" will result in a map with `email` set and `name` as `undefined`.
+If the format is invalid, the function will abort with an error.
+""".
+-spec parse_authors(Authors) -> Result
+    when Authors :: binary(),
+         Result  :: [#{name => binary(), email => binary()}] | no_return().
+parse_authors(Authors) ->
+    AuthorEntries = string:split(Authors, ",", all),
+    lists:map(fun parse_author/1, AuthorEntries).
+
+parse_author(AuthorEntry) ->
+    case string:split(AuthorEntry, ":", all) of
+        [Name, Email] when Name =/= [] andalso Email =/= [] ->
```

Comment:
It's not entirely safe to check string equality like this.
It might be true that if we consider the entire call chain that leads up to this, then it's guaranteed that an empty string will be a `[]`, but in general an `[]` is an `[[]]` is an `""` (and others) when talking about strings, even though they are not `=:=`.
So you should use the `string:equals/2` function when comparing strings.
Especially because this function also does normalization.